pygfx/helpers/_box.py

```diff
@@ -4,14 +4,29 @@
 
 
 class BoxHelper(Line):
-    """A line box object. Commonly used to visualize bounding boxes.
+    """Generate a box-shaped wireframe.
+
+    Commonly used to visualize bounding boxes.
+
+    Parameters
+    ----------
+    size : float
+        The length of the box' edges in local space.
+    thickness : float
+        The thickness of the lines in (onscreen) pixels.
+    color : Color
+        The color of the box.
+
+    Examples
+    --------
+    .. minigallery:: pygfx.BoxHelper
 
-    Parameters:
-        size (float): The length of the box' edges (default 1).
-        thickness (float): the thickness of the lines (default 1 px).
-        color (Color): the color of the box.
     """
 
+    # @almarklein: Should we move this to geometry._box? It clearly has a
+    # usecase as a aabb helper, but seems to have merit outside of this context,
+    # too.
```

Comment:
I don't have a strong opinion on the module that this is in, but its still be a helper, so this module makes sense? 😃

Reply:
We could call it `WireframeBox` and then the name "Helper" disappears ;). I mainly think that a box wireframe can do more than just help draw a bounding box.

(also while on the topic of wireframes, shouldn't it be possible to write a shader that draws a wireframe instead of drawing faces, removing the need for explicit wireframe objects?)

Reply:
> We could call it `WireframeBox` and then the name "Helper" disappears ;). I mainly think that a box wireframe can do more than just help draw a bounding box.

I'm not convinced yet. It still feels like a helper to me 😄. If you take your position to the extreme, you could argue that all objects in the helper submodule should be first class world objects. In the end it does not matter much where they live. Let's discuss in a new issue?

> (also while on the topic of wireframes, shouldn't it be possible to write a shader that draws a wireframe instead of drawing faces, removing the need for explicit wireframe objects?)

Yes, we have that! It's the `MeshMaterial.wireframe` property. But you'll see wires for all the edges of the triangle, so it serves a different purpose.

Reply:
The qualitative difference between geometry and helper is that helpers are WorldObjects and geometries are routines that produce geometry which can be attached to WorldObjects.

In this specific case, if anything, you could extract the parts of code that generate the box's geometry to a function in the geometry subpackage. But the box helper will remain.

Reply:
> Yes, we have that! It's the MeshMaterial.wireframe property. But you'll see wires for all the edges of the triangle, so it serves a different purpose.

Ah right, then it isn't equivalent to drawing a wireframe box as we do here.

> WorldObjects typically used for visual debugging, while the geometry package holds routines that produce geometry which can be attached to WorldObjects

Ah, that's the piece I was missing. It will be good to have this info in the docs of the helpers module, because it makes it clear why the box helper is relevant here. I'll go and update the docstring.